### ISSUE TITLE
jsx-a11y eslint plugin reports various lint issues with paginator

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#1689]
# Bug Report

## Description
The following lint errors show up for the paginator component after updating eslint / eslint-jsx-a11y plugin:

* jsx-a11y/no-static-element-interactions
* jsx-a11y/anchor-is-valid
* jsx-a11y/no-noninteractive-tabindex

## Steps to Reproduce

1. Run eslint with eslint v4+ and jsx-a11y plugin v6.1.0+

## Expected Behavior
Lint errors are addressed and resolved

## Possible Solution
Currently, the pagination component uses anchor elements without href attributes. Switching over to using the button component could resolve these lint issues.

## Environment

* Component Name and Version: terra-paginator v1.0.0+

**Comment by nramamurth** [comment]
I have an initial rough draft to resolve this issue - https://github.com/cerner/terra-core/commit/4e9bba3187f3f902a567d2a38fea5fb81ce19a94. 

Overall, I believe the following are the changes that I believe that need to be made while translating all the anchor tags to be of type `Button`.

1. **Translations**

The text 'First', 'Next', 'Previous', 'Last' need to be internationalised, methinks. 

https://github.com/cerner/terra-core/blob/25696dcf0fb49d22356219252f13ef0ef83b98e7/packages/terra-paginator/src/Paginator.jsx#L150-L152

https://github.com/cerner/terra-core/blob/25696dcf0fb49d22356219252f13ef0ef83b98e7/packages/terra-paginator/src/Paginator.jsx#L163-L165

2. **Transformation of First anchor tag to First button.**

```diff
- <a
+ <Button
      aria-disabled={selectedPage === 1}
-     aria-label="first"
+     isDisabled={selectedPage === 1}
-     className={cx(['nav-link', 'left-controls', selectedPage === 1 && 'is-disabled'])}
+     className={cx(['nav-link',  selectedPage === 1 && 'is-disabled'])}
      tabIndex={selectedPage === 1 ? null : '0'}
      onClick={this.handlePageChange(1)}
      onKeyDown={this.handleOnKeyDown(1)}
+     text="First" // Internationalised.
-  >
+ />
-  First
- </a>

```

> Similar changes need to be made for Last button.

3. **Transformation of Previous anchor tag to Previous button.**

```diff
- <a
+ <Button
     aria-disabled={selectedPage === 1}
-    aria-label="previous"
-    className={cx(['nav-link', 'left-controls', 'previous', selectedPage === 1 && 'is-disabled'])}
+    className={cx(['nav-link', 'previous', selectedPage === 1 && 'is-disabled'])}
     tabIndex={selectedPage === 1 ? null : '0'}
     onClick={this.handlePageChange(previousPageIndex)}
     onKeyDown={this.handleOnKeyDown(previousPageIndex)}
+    text="Previous" // Internationalised
+    icon={<IconPrevious />}
+    isDisabled={selectedPage === 1}
-  >
+ />
-          <span className={cx('icon')} />
- Previous
- </a>
```

> - Similar changes need to be made for Next button but with `isReversed` prop so the `<IconNext />` appears to the right of the text.
> - Add `isIconOnly` to these buttons for the "Reduced Paginator".

4. **Changes in `handlePageChange` and `handleOnKeyDown`:**

```diff
 if (Number.isNaN(Number(index))) {
-  this.props.onPageChange(event.currentTarget.attributes['aria-label'].value);
+  this.props.onPageChange(event.currentTarget.textContent);

   return false;
 }
```

5. **Evaluate CSS Changes:** 

https://github.com/cerner/terra-core/commit/4e9bba3187f3f902a567d2a38fea5fb81ce19a94#diff-dbd3b75a65ed5bc7cbb843b08ef6e93c

I have removed most of the CSS that were not needed imo. @neilpfeiffer Could you please re-evaluate them for me?

@bjankord @neilpfeiffer @ryanthemanuel @ShettyAkarsh Please let me know your thoughts.

**Comment by nramamurth** [comment]
Upon further initial refactoring, I ran into an issue with how the methods [handlePageChange](https://github.com/cerner/terra-core/blob/master/packages/terra-paginator/src/Paginator.jsx#L51) and [handleOnKeyDown](https://github.com/cerner/terra-core/blob/master/packages/terra-paginator/src/Paginator.jsx#L71) handle navigation in a [paginator w/o pages](https://github.com/cerner/terra-core/blob/master/packages/terra-paginator/src/terra-dev-site/doc/example/PaginatorNoPagesExample.jsx). 

https://github.com/cerner/terra-core/blob/25696dcf0fb49d22356219252f13ef0ef83b98e7/packages/terra-paginator/src/Paginator.jsx#L76-L80

I noticed that for a default `Button`, the `aria-label` supplied by the consumer is nuked which causes an issue in the above code because the supplied `aria-label` to the buttons 'Next' and 'Previous' are nuked. 

https://github.com/cerner/terra-core/blob/25696dcf0fb49d22356219252f13ef0ef83b98e7/packages/terra-button/src/Button.jsx#L246

I'd believe this is a bug in `Button` but I am curious if there was a conscious decision made during the design of the `Button` component. 

@bjankord @neilpfeiffer What are your thoughts?

**Comment by nramamurth** [comment]
**Updates:** 

1. Offline, Neil confirmed that this issue mentioned in https://github.com/cerner/terra-core/issues/1689#issuecomment-483123764 should be fixed. 

2. Neil was also wondering if `Paginator` will need its own buttons since the designs that we have to achieve in the `Paginator` component may not (or do not) completely match designs that we have been given in the button component and consistent in all themes (Clinical Theme, MPage Theme, Default Terra theme). 

To evaluate the styles further on all themes, I have an "almost working" branch - https://github.com/cerner/terra-core/compare/paginator-lint.

@neilpfeiffer Please feel free to update it if I have quoted you wrong :)

**Comment by bjankord** [comment]
Yeah, I'd recommend we don't use terra-button for this. I looked through the design UX has provided and it looks like this will need it's own button styles. We should use an button HTML element and apply pagination button styles to that.

**Comment by nramamurth** [comment]
@neilpfeiffer Here's a new branch with changes using HTML button elements - https://github.com/cerner/terra-core/compare/paginator-buttons. 

I think it is _important_ to point out that the current behaviour of the click action on _Paginator_ is not consistent across all browsers: 
- Chrome: the page number links **do not**  persist a border and a shadow when they are clicked.
- Safari & Firefox: the page number links persist a border and a shadow when they are clicked.

**Safari:** 

![safari](https://user-images.githubusercontent.com/10090859/56195888-6edc6380-6053-11e9-966a-89ff170939f7.png)

**Firefox:**

<img width="772" alt="firefox" src="https://user-images.githubusercontent.com/10090859/56195902-77349e80-6053-11e9-90ef-f29fef164eab.png">

**Chrome:** 

<img width="768" alt="chrome" src="https://user-images.githubusercontent.com/10090859/56195917-7dc31600-6053-11e9-9acd-39ef4244c30d.png">

With buttons, this behaviour has changed. Please let me know the expected behaviour across all browsers so I could address the issue accordingly.

**Comment by nramamurth** [comment]
@neilpfeiffer With new button implementation on IE 11 🤯💥

![Screen Shot 2019-04-17 at 11 11 54 AM](https://user-images.githubusercontent.com/10090859/56264160-35fbc780-6103-11e9-8047-44b81fa54f93.png)

**Comment by ryanthemanuel** [comment]
Let's make sure we're keeping https://github.com/cerner/terra-core/issues/1689 in mind with this as well.

**Comment by bjankord** [comment]
Cross Post from #1689
> For every page, it should say “Goto Page #”
For the current page it should say “Current Page, Page #”
First, Previous, next and last stay the same
These rule apply for both paginates

https://medium.com/@jessebeach/beware-smushed-off-screen-accessible-text-5952a4c2cbfe
